Sources/Datadog/DatadogInternal/Concurrency/Concurrency.swift

```diff
@@ -0,0 +1,56 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+/// A property wrapper using a fair, POSIX conforming reader-writer lock for atomic
+/// access to the value.
+///
+/// The wrapper is a class to prevent copying the lock, it creates and initilaizes a `pthread_rwlock_t`.
+/// An additional method `mutate` allow to safely mutate the value within a closure.
```

Comment:
Struggled a bit to understand the purpose of `mutate`. Wouldn't that make it cleaner?
```suggestion
/// An additional method `mutate` allow to safely mutate the value in-place (to read it 
/// and write it while obtaining the lock only once).
```